### ISSUE TITLE
Modules as implementations of protocols

### ISSUE THREAD

**Issue by simwr872** [microsoft/pyright#1341]
**Describe the bug**
A module object compatible with a defined protocol is not accepted where a protocol is expected.

**To Reproduce**
See minimal code example below.

**Expected behavior**
Accept any module compatible with the defined protocol as stated in [https://www.python.org/dev/peps/pep-0544/#modules-as-implementations-of-protocols](https://www.python.org/dev/peps/pep-0544/#modules-as-implementations-of-protocols).

**Screenshots or Code**
```python
import sqlite3
from typing import Protocol

class PEP249Module(Protocol):
    paramstyle: str

module: PEP249Module = sqlite3
```
Running pyright results in the following report:
```
  7:24 - error: Expression of type "Module("sqlite3")" cannot be assigned to declared type "PEP249Module"
    Type "Module("sqlite3")" cannot be assigned to type "PEP249Module" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command-line using version `1.1.99`.

**Comment by erictraut** [comment]
Yes, this is correct. Modules cannot be used to satisfy protocols. I confirmed this behavior is consistent with other type checkers as well.

**Comment by erictraut** [comment]
Upon closer reading of PEP 544, it looks like this should be allowed.

https://www.python.org/dev/peps/pep-0544/#modules-as-implementations-of-protocols

**Comment by erictraut** [comment]
It looks like this was never implemented in mypy or other Python type checkers. This issue has been open for over two years now: https://github.com/python/mypy/issues/5018.

**Comment by erictraut** [comment]
This will be implemented in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is implemented in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
